### PR TITLE
:seedling: bump markdownlint-cli2 to v0.12.0

### PR DESCRIPTION
**Pull request by tuminoid**
Manual cherry-pick of #2006

### CODE REVIEW

File: hack/markdownlint.sh [hunk 1]
```diff
@@ -15,6 +15,6 @@ else
         --volume "${PWD}:/workdir:ro,z" \
         --entrypoint sh \
         --workdir /workdir \
-        docker.io/pipelinecomponents/markdownlint-cli2:0.9.0@sha256:71370df6c967bae548b0bfd0ae313ddf44bfad87da76f88180eff55c6264098c \
+        docker.io/pipelinecomponents/markdownlint-cli2:0.12.0@sha256:a3977fba9814f10d33a1d69ae607dc808e7a6470b2ba03e84c17193c0791aac0  \
```
Comment (kashifest) [inline hack/markdownlint.sh:18]:
```suggestion
        docker.io/pipelinecomponents/markdownlint-cli2:0.12.0@sha256:a3977fba9814f10d33a1d69ae607dc808e7a6470b2ba03e84c17193c0791aac0 \
```

**Review by kashifest — COMMENTED** [review]
/approve

File: hack/markdownlint.sh (hunk 1 above)
Comment (tuminoid) [reply hack/markdownlint.sh]: Fixed

**Review by adilGhaffarDev — COMMENTED** [review]
/lgtm

**Pull request merged**